### ISSUE TITLE
String exceptions will be removed in python 2.6

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2762]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#2762|
|Type|defect|
|Created|2007-08-01 07:53:25Z|
|Branch|https://github.com/twisted/twisted/tree/string-exceptions-2762|

After being deprecated in python 2.5, string exceptions will cause an error in python 2.6.

There aren't many consequences, mainly a failing test: ```twisted.test.test_failure.FailureTestCase.testStringExceptions```

<details><summary>Searchable metadata</summary>

```
trac-id__2762 2762
type__defect defect
reporter__therve therve
priority__low low
milestone__Python_2_6 Python-2.6
branch__branches_string_exceptions_2762 branches/string-exceptions-2762
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1185954805000000 1185954805000000
changetime__1209231189000000 1209231189000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

It's also related to [#2063](https://github.com/twisted/twisted/issues/2063).

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve commented|
|-|-|

(In [23436]) Branching to 'string-exceptions-2762'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [23437]) Add 2 skips for >= 2.6, and a suppress for 2.5.

Refs [#2762](https://github.com/twisted/twisted/issues/2762)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve removed owner|
|-|-|

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @therve|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_8"></a>|@therve set status to `closed`|
|-|-|

(In [23438]) Merge string-exceptions-2762

Author: therve
Reviewer: exarkun
Fixes [#2762](https://github.com/twisted/twisted/issues/2762)

Skip tests for string exceptions under Python 2.6.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
